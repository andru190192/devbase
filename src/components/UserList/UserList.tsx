import React from 'react';
import {Box, Button} from 'native-base';

export interface UserListProps {
  users: {id: number; name: string}[];
  onHandlerUser: (name: string) => string;
}

function UserList({users, onHandlerUser}: UserListProps): JSX.Element {
  return (
    <Box alignItems="center" mt={5}>
      {users.map(user => {
        return (
          <Button key={user.id} my={2} onPress={() => onHandlerUser(user.name)}>
            {user.name}
          </Button>
        );
      })}
    </Box>
  );
}

export default UserList;
