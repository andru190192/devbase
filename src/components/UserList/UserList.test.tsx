import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import {NativeBaseProvider} from 'native-base';
import UserList, {UserListProps} from '../UserList/UserList';

const onHandlerUserMock = jest.fn();

const props: UserListProps = {
  users: [{id: 1, name: 'Andres Guachisaca'}],
  onHandlerUser: onHandlerUserMock,
};

const inset = {
  frame: {x: 0, y: 0, width: 0, height: 0},
  insets: {top: 0, left: 0, right: 0, bottom: 0},
};

describe('', () => {
  test('should render correctly', () => {
    const component = render(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <UserList {...props} />
      </NativeBaseProvider>,
    );

    const userName = screen.getByText('Andres Guachisaca');

    expect(component).toMatchSnapshot();
    expect(userName).toBeTruthy();
  });

  test('should call onHandlerUser when tap on user', () => {
    render(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <UserList {...props} />
      </NativeBaseProvider>,
    );

    const userName = screen.getByText('Andres Guachisaca');
    fireEvent.press(userName);

    expect(onHandlerUserMock).toHaveBeenCalledWith('Andres Guachisaca');
  });
});
