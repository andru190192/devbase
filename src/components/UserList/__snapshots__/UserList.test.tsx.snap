// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "alignItems": "center",
        "marginTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      colorScheme="primary"
      dataSet={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#0891b2",
          "borderRadius": 4,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginBottom": 8,
          "marginTop": 8,
          "paddingBottom": 10,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 10,
        }
      }
    >
      <View
        dataSet={{}}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
        test={true}
      >
        <View
          dataSet={{}}
          style={{}}
        >
          <Text
            dataSet={{}}
            style={
              {
                "backgroundColor": undefined,
                "color": "#fafafa",
                "fontFamily": undefined,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 21,
                "textDecorationLine": undefined,
              }
            }
          >
            Andres Guachisaca
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
